src/updatemanager.rs

```diff
@@ -0,0 +1,355 @@
+/*
+ * This file is part of Edgehog.
+ *
+ * Copyright 2022 SECO Mind Srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use log::{debug, error, info, warn};
+use serde::{Deserialize, Serialize};
+use zbus::export::futures_util::StreamExt;
+
+use crate::error::DeviceManagerError;
+use crate::rauc::RaucProxy;
+
+#[derive(Serialize, Deserialize, Debug)]
+struct PersistentState {
+    uuid: String,
+    slot: String,
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum OtaError {
+    #[error("Bundle download failed")]
+    Network,
+    #[error("Deployment failed")]
+    Deploy,
+    #[error("OTA failed")]
+    Failed,
+}
+
+pub struct UpdateManager<'a> {
```

Comment:
@bettio What do you think about this name? IMHO it should be called something like `OTAHandler`